int32 mode_index
---
bool status